### PR TITLE
Add docs for public services

### PR DESCRIPTION
**Pull request by yangyansong-adbe**

### CODE REVIEW

File: docs/Services/README.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    override fun warning(tag: String, message: String) {}
+    override fun error(tag: String, message: String) {
+        Log.e("ErrorLogger", message)
+    }
+}
+```
+
+In the code snippet above, we have a class that implements `Logging` and provides simple implementation for the single required API.
+
+## Overriding a Service
```
Comment (praveek) [inline docs/Services/README.md:93]: Can you also add a note saying services should only be overridden before initializing the SDK?

File: docs/Services/README.md [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+As we saw above, implementing the `Logging` interface was quite simple, but how do we get the entire SDK to take advantage of this new service in place of the default implementation?
+
+We can do this by setting the `loggingService` on the shared `ServiceProvider`, used by the entire SDK.
+
+> Services overriding should be done before the Android SDK is initiated, then the SDK can switch to use the overridden services.
```
Comment (praveek) [inline docs/Services/README.md:99]: initialized

Comment (praveek) [reply docs/Services/README.md:99]: Can you reword this? 
> For the Android SDK to use overridden services, Services overriding should be done before the SDK is initialized.

File: docs/Services/URLOpening.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# URLOpening
+
+## Overview
+
+The `URLOpening` service provides shared functionality for opening URLs. Overriding this service can be helpful to determine when the SDK will attempt to open a given URL.
+
+## Declaration
+
+##### `public protocol URLOpening`
```
Comment (praveek) [inline docs/Services/URLOpening.md:9]: Can you change this to `interface`

Comment (yangyansong-adbe) [reply docs/Services/URLOpening.md:9]: Oh, the whole page should be removed. As we discussed, we don't need to show any details for the`UIService`.

**Review by praveek — APPROVED** [review]

**Pull request merged**